### PR TITLE
executer: collect and write analyze at the same time

### LINKED ISSUE #47339: Collect and Write Analyze Data Simultaneously
## Enhancement
When dealing with large datasets, we often need to first collect the data and then analyze it. However, these two steps are currently separate, which can lead to performance issues, especially in memory-limited situations.

I suggest we refactor the "executer" to allow for simultaneous data collection and analysis. This could improve performance and reduce memory usage. Specifically, we could perform some basic analysis operations, such as counting, summing, etc., while collecting the data. This way, we could get faster results without increasing memory usage.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #47339

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/analyze.go [hunk 1]
```diff
@@ -115,6 +116,14 @@ func (e *AnalyzeExec) Next(ctx context.Context, _ *chunk.Chunk) error {
 	for i := 0; i < concurrency; i++ {
 		e.wg.Run(func() { e.analyzeWorker(taskCh, resultsCh) })
 	}
+	pruneMode := variable.PartitionPruneMode(sessionVars.PartitionPruneMode.Load())
+	// needGlobalStats used to indicate whether we should merge the partition-level stats to global-level stats.
+	needGlobalStats := pruneMode == variable.Dynamic
+	globalStatsMap := make(map[globalStatsKey]globalStatsInfo)
+	g, _ := errgroup.WithContext(context.Background())
```
Comment (winoros) [inline executor/analyze.go:123]: Should we use the ctx passed from the `Next()`?

Comment (hawkingrei) [reply executor/analyze.go]: Done

**Review by qw4990 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**